src/products/entities/product.repository.ts

```diff
@@ -0,0 +1,12 @@
+import {EntityRepository, Repository} from "typeorm";
+import {Product} from "./product.entity";
+
+@EntityRepository(Product)
+export class ProductRepository extends Repository<Product> {
+    findByIdx(id: number) {
```

Comment:
Medium: The base repository class should have a findbyId method (or simply find where you can pass the ID). We try to avoid using the query builder unless in cases where we need to do complex joins, projections, etc.

Reply:
I actually used findById, I think that "x" was a last-minute typo